daq/local_state_collector.py

```diff
@@ -38,6 +43,18 @@ def get_process_state(self, extended=True):
 
         return self._process_state
 
+    def _get_target_processes(self):
+        """Get target processes"""
+        procs = {}
+        for proc in psutil.process_iter():
+            for target_name, (target_cmd, target_regex) in self._target_procs.items():
+                if proc.name() == target_cmd:
+                    cmd_line_str = ''.join(proc.cmdline())
+                    if re.search(target_regex, cmd_line_str):
+                        procs[target_name] = proc
+                        break
```

Comment:
What does this do if the regex matches two entries in the process table?  This should be an obvious error indicated in the result somehow, so it's very obvious if things are broken.  Whatever regex we have now may work for the current setup, but we need it to be future-proof in the face of changes on the controller system.

Reply:
The most precise way is to fully compare the command line fields, i.e., every filed must match. However, the matching fields vary on different setups. For example, some setup is using python3.6, some setup is using python3.7. Some is using venv, some is not. Some is running with specific flag, some is not. For now, regex comparison is the easiest and universal way to identify each process.

I think I can add a TODO and print a error message if duplication is found.